src/index.js

```diff
@@ -22,12 +24,17 @@ export default class SpoorClient {
 		this.category = category;
 		this.product = product;
 		this.req = req;
+		this.cookies = cookies;
+		this.ua = ua;
 		this.shouldSubmitEvent = submitIf;
 		this.apiKey = apiKey;
 		this.inTestMode = inTestMode;
 	}
 
-	submit ({source=this.source, category=this.category, req=this.req, apiKey=this.apiKey, product=this.product, action, context}={}) {
+	submit ({source=this.source, category=this.category, req=this.req, cookies=this.cookies, ua=this.ua, apiKey=this.apiKey, product=this.product, action, context}={}) {
```

Comment:
Actually, I think the split between constructor instance vars and transitory params to `submit` should be much clearer. `source`, `req`, `cookies`, `ua`, `product`, `apiKey`, `submitIf` and `inTestMode` belong in the constructor. `submit` should only accept `category`, `action` and `context`. But I need to know what upstream clients are sending to the API to do that (or just blindly release a breaking change!).